### PR TITLE
fix(sast): fix sast report format

### PR DESCRIPTION
**Pull request by achiar99**
**By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

[//]: # "
    # PR Title
    Be aware that we use the title to create changelog automatically and therefore only allow specific prefixes
    - break:    to indicate a breaking change, this supersedes any of the types
    - feat:     to indicate new features or checks
    - fix:      to indicate a bugfix or handling of edge cases of existing checks
    - docs:     to indicate an update to our documentation
    - chore:    to indicate adjustments to workflow files or dependency updates
    - platform: to indicate a change needed for the platform
    Additionally a scope is needs to be added to the prefix, which indicates the targeted framework, in doubt choose 'general'.
    #    
    Allowed prefixs:
    ansible|argo|arm|azure|bicep|bitbucket|circleci|cloudformation|dockerfile|github|gha|gitlab|helm|kubernetes|kustomize|openapi|sast|sca|secrets|serverless|terraform|general|graph|terraform_plan|terraform_json
    #
    ex.
    feat(terraform): add CKV_AWS_123 to ensure that VPC Endpoint Service is configured for Manual Acceptance
"

## Description

Fix sast report format for bc integration

*Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.*

Fixes # (issue)

### Description
*Include a description of what makes it a violation and any relevant external links.*

### Fix
*How does someone fix the issue in code and/or in runtime?*

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my feature, policy, or fix is effective and works
- [ ] New and existing tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by arielkru — APPROVED** [review]

File: checkov/common/bridgecrew/platform_integration.py [hunk 1]
```diff
@@ -554,6 +586,8 @@ def persist_scan_results(self, scan_reports: list[Report]) -> None:
         dpath.merge(reduced_scan_reports, checks_metadata_paths)
         persist_checks_results(reduced_scan_reports, self.s3_client, self.bucket, self.repo_path)
```
Comment (mikeurbanski1) [inline checkov/common/bridgecrew/platform_integration.py:587]: Do the sast reports need to be removed from `reduced_scan_reports`?

File: checkov/common/bridgecrew/platform_integration.py [hunk 2]
```diff
@@ -554,6 +586,8 @@ def persist_scan_results(self, scan_reports: list[Report]) -> None:
...
 
+        self.persist_sast_checks_results(scan_reports)
```
Comment (mikeurbanski1) [inline checkov/common/bridgecrew/platform_integration.py:589]: Make sure you strip any code from the report if `self.on_prem` is true (see https://github.com/bridgecrewio/checkov/blob/e5e2c453edc4addbc7660b770032d0af33c0f5aa/checkov/common/bridgecrew/platform_integration.py#L551 ).

Please also check that `enrich_and_persist_checks_metadata` will remove any code if it is stored in a different way than the current reports (`code_block`).

File: checkov/main.py [hunk 3]
```diff
@@ -509,6 +510,7 @@ def run(self, banner: str = checkov_banner, tool: str = checkov_tool, source_typ
 
             if self.config.directory:
                 exit_codes = []
+                bc_integration.scan_dir = self.config.directory
```
Comment (mikeurbanski1) [inline checkov/main.py:513]: Does this need to be set for `-f`?

File: checkov/common/bridgecrew/platform_integration.py (hunk 1 above)
Comment (achiar99) [reply checkov/common/bridgecrew/platform_integration.py:587]: Yes. We persist it here - `persist_sast_checks_results`

**Review by mikeurbanski1 — APPROVED** [review]

**Review by pazbechor — APPROVED** [review]
Github\s V

**Pull request merged**